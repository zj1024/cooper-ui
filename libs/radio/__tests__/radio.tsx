import * as React from 'react'
import * as renderer from 'react-test-renderer'
import Radio from '../index'

const options = [
  {
    label: 'Harrison',
    value: 'Harrison',
    disabled: true,
    checked: true,
  },
  {
    label: 'Jack',
    value: 'Jack',
    checked: true,
  },
  {
    label: 'Barry',
    value: 'Barry',
  },
]

const fn = jest.fn()

describe('- Radio component', () => {
  it(`accept mode vertical`, () => {
    const component = renderer.create(<Radio mode="vertical" options={options} onChange={fn} />)
    const tree: any = component.toJSON()
    expect(tree.props.className.includes('coo-radio-vertical')).toBeTruthy()
  })

  it('exist', () => {
    const component = renderer.create(<Radio options={options} onChange={fn} />)
    const tree: any = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`accept name`, () => {
    const component = renderer.create(<Radio name="test" options={options} onChange={fn} />)
    const tree: any = component.toJSON()
    const children = tree.children.map((d: any) => d.children)
    // 找到input标签上的name是否等于test
    const notFindNameLength = children
      .map((d: any) => {
        return d[1].props.name === 'test'
      })
      .filter((d: any) => d === false).length
    expect(notFindNameLength).toEqual(0)
  })
})
