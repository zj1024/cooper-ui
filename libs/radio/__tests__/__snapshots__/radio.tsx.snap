// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`- Radio component exist 1`] = `
<div
  className="coo-radio coo-radio-horizontal"
>
  <label
    className="coo-radio-label coo-radio-label-disabled"
    style={Object {}}
  >
    <span
      className="coo-radio-inner"
    />
    <input
      checked={false}
      className="coo-radio-input"
      disabled={true}
      onChange={[Function]}
      type="radio"
      value="Harrison"
    />
    <span
      className="coo-radio-label-text"
    >
      Harrison
    </span>
  </label>
  <label
    className="coo-radio-label coo-radio-label-active"
    style={Object {}}
  >
    <span
      className="coo-radio-inner"
    />
    <input
      checked={true}
      className="coo-radio-input"
      disabled={false}
      onChange={[Function]}
      type="radio"
      value="Jack"
    />
    <span
      className="coo-radio-label-text"
    >
      Jack
    </span>
  </label>
  <label
    className="coo-radio-label coo-radio-label-last"
    style={Object {}}
  >
    <span
      className="coo-radio-inner"
    />
    <input
      checked={false}
      className="coo-radio-input"
      disabled={false}
      onChange={[Function]}
      type="radio"
      value="Barry"
    />
    <span
      className="coo-radio-label-text"
    >
      Barry
    </span>
  </label>
</div>
`;
